src/pages/index.js

```diff
@@ -0,0 +1,357 @@
+      previewModalImageEl.src = data.link;
+      previewModalImageEl.alt = data.name;
+    });
+  
+    cardDeleteBtn.addEventListener("click", () => 
+      handleDeleteCard(cardElement, data._id, cardDeleteBtn));
+  
+    return cardElement;
+  }
+  
+  function handleDeleteCard(cardElement, cardId, cardDeleteBtn) {
+    selectedCard = cardElement;
+    selectedCardId = cardId;
+    openModal(deleteModal);
+  }
+  
+  function handleDeleteSubmit(evt) {
+    evt.preventDefault();
+  
+    setButtonText(avatarDeletetBtn, true, "Delete", "Deleting...");
+  
+    api.deleteCard(selectedCardId)
+      .then(() => {
+        selectedCard.remove();
+        closeModal(deleteModal);
+      })
+      .catch(console.error)
+      .finally(() => {
+        setButtonText(avatarDeletetBtn, false, "Delete");
+      });
+  }
+  
+  function handleEditFormSubmit(evt) {
+    evt.preventDefault();
+    const profileSubmitBtn = evt.submitter;
+    setButtonText(profileSubmitBtn, true, "Save", "Saving...");
+  
+    const updatedUserInfo = {
+      name: editModalNameInput.value,
+      about: editModalAboutInput.value
+    };
+  
+    api.editUserInfo(updatedUserInfo)
+      .then((data) => {
+        profileName.textContent = data.name;
+        profileDescription.textContent = data.about;
+        closeModal(editModal);
+      })
+      .catch((error) => {
+        console.error(error);
+      })
+      .finally(() => {
+        profileSubmitBtn.disabled = false;
+        profileSubmitBtn.classList.remove("modal__submit-btn_disabled");
+        setButtonText(profileSubmitBtn, false, "Save");
+        editFormElement.reset();
+      });
+  }
+  
+  function toggleSubmitButton() {
+    const isInputValid = cardNameInput.value.trim() !== "" && cardLinkInput.value.trim() !== "";
+    cardSubmitBtn.disabled = !isInputValid; 
+  }
+  
+  function handleAddCardSubmit(evt) {
+    evt.preventDefault();
+    setButtonText(cardSubmitBtn, true, "Save", "Saving...");
+  
+    const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  
+    api.createCard(inputValues)
+      .then((data) => {
+        const cardElement = getCardElement(data);
+        cardsList.prepend(cardElement);
+        closeModal(cardModal);
+      })
+      .catch((error) => {
+        console.error(error);
+      })
+      .finally(() => {
+        cardSubmitBtn.disabled = false;
+        cardSubmitBtn.classList.remove("modal__submit-btn_disabled");
+        setButtonText(cardSubmitBtn, false, "Save");
+        cardForm.reset();
+      });
+  }
+  
+  function handleAvatarSubmit(evt) {
+    evt.preventDefault();
+    setButtonText(avatarSubmitBtn, true, "Save", "Saving...");
+  
+    api.editAvatarInfo(avatarInput.value)
+      .then((data) => {
+        avatarImage.src = data.avatar;
+        closeModal(avatarModal);
+      })
+      .catch(console.error)
+      .finally(() => {
+        setButtonText(avatarSubmitBtn, false, "Save");
+        avatarForm.reset();
```

Comment:
` avatarForm.reset();` should be in `.then` in line 256

* Please, **disable** the `avatar` submit button after submission in `.then` as well https://snipboard.io/7fwgxu.jpg 

You'll have this code here:

```js
function handleAvatarSubmit(evt) {
  evt.preventDefault();
  setButtonText(avatarSubmitBtn, true, "Save", "Saving...");

  api
    .editAvatarInfo(avatarInput.value)
    .then((data) => {
      avatarImage.src = data.avatar;
      closeModal(avatarModal);
      avatarForm.reset();  // reset in `.then`
      disabledButton(avatarSubmitBtn, settings);  // disable the avatar submit button
    })
    .catch(console.error)
    .finally(() => {
      setButtonText(avatarSubmitBtn, false, "Save");
    });
}
```